File: 3-2.py
#!/usr/bin/env python

from functools import reduce


def priority(item):
    if ord(item) in range(ord('a'), ord('z') + 1):
        return ord(item) - 96
    if ord(item) in range(ord('A'), ord('Z') + 1):
        return ord(item) - 64 + 26


rucksacks = [row.strip() for row in open('3.input').readlines()]

p = 0
for base in range(0, len(rucksacks), 3):
    badge = set(rucksacks[base]) & \
        set(rucksacks[base + 1]) & \
        set(rucksacks[base + 2])
    p += priority(badge.pop())

print(p)

File: 3-2.rb
#!/usr/bin/env ruby

require 'set'

def priority(item)
  case item
  when 'a'..'z'
    item.ord - 96
  when 'A'..'Z'
    item.ord - 64 + 26
  end
end

p = 0
File.read('3.input').lines.map(&:strip).each_slice(3) do |group|
  badge = Set.new(group[0].chars) &
          Set.new(group[1].chars) &
          Set.new(group[2].chars)
  p += priority(badge.to_a[0])
end
print p, "\n"
